src/sentry/spans/buffer.py

```diff
@@ -153,6 +154,13 @@ def __init__(self, assigned_shards: list[int]):
         self.assigned_shards = list(assigned_shards)
         self.add_buffer_sha: str | None = None
         self.any_shard_at_limit = False
+        # Reuse compressor/decompressor objects for better performance
+        compression_level = options.get("spans.buffer.compression.level")
+        if compression_level == -1:
+            self._zstd_compressor = None
+        else:
+            self._zstd_compressor = zstandard.ZstdCompressor(level=compression_level)
```

Comment:
Could we either instantiate or override the compressor for each batch separately, so we can change the compression level at runtime without a consumer restart? This would help with enabling this for the first time via options-automator.

Reply:
done